web3.js/src/transaction/legacy.ts

```diff
@@ -805,7 +805,7 @@ export class Transaction {
     for (let i = 0; i < signatureCount; i++) {
       const signature = byteArray.slice(0, SIGNATURE_LENGTH_IN_BYTES);
       byteArray = byteArray.slice(SIGNATURE_LENGTH_IN_BYTES);
-      signatures.push(bs58.encode(Buffer.from(signature)));
+      signatures.push(bs58.encode(Uint8Array.from(signature)));
```

Comment:
We can go straight into `bs58.encode()` as an array of numbers!
```suggestion
      signatures.push(bs58.encode(signature));
```